neural_compressor/compression/pruner/pruners.py

```diff
@@ -96,27 +96,28 @@ def get_pruner(config, modules):
         assert False, f"does not support {name}, currently only support {parse_valid_pruner_types()}"
     return PRUNERS[name](config, modules)
 
-def model_slim(model, round_multiplier=0):
+def model_slim(model, dataloader=None, round_multiplier=0):
     """Slim the sparse model automatically."""
-    model = model_slim_ffn2(model, round_multiplier)
-    model = model_slim_mha(model)
+    model = model_slim_ffn2(model, dataloader, round_multiplier)
+    model = model_slim_mha(model, dataloader)
     return model
 
-def model_slim_ffn2(model, round_multiplier=0):
+def model_slim_ffn2(model, dataloader = None, round_multiplier=32):
```

Comment:
better move these three functions to model_slim folder

Reply:
created a auto_slim under model_slim to store api functions